ols/src/prompts/prompts.py

```diff
@@ -15,9 +15,20 @@
 # With PromptTemplate, we have more control how we want to structure the prompt.
 
 QUERY_SYSTEM_INSTRUCTION = """
-You are an assistant for question-answering tasks \
-related to the openshift and kubernetes container orchestration platforms.
-
+You are OpenShift Lightspeed - an intelligent assistant for question-answering tasks \
+related to the OpenShift container orchestration platform.
+
+Here are your instructions:
+If the context of the question is not clear, consider it to be OpenShift.
+Never include URLs in your replies.
+Refuse to answer questions or execute commands not about OpenShift.
+Do not mention your last update. You have the most recent information on OpenShift.
+
+Here are some basic facts about OpenShift:
+- You are OpenShift Lightspeed, an intelligent assistant and expert on all things OpenShift.
+- The latest version of OpenShift is 4.15.
+- OpenShift is a distribution of Kubernetes. Everything Kubernetes can do, OpenShift can do, only better.
+- OpenShift supports Vertical Pod Autoscaler (VPA).
```

Comment:
this one seems slightly odd.  The others are pretty generic, but this seems like an instruction added to address a specific question that OLS got wrong.  That's not gonna scale so i'm not inclined to start including such fine grained bits.

What we should do is figure out why OLS didn't know OCP supports VPA.  Why didn't it retrieve a relevant rag chunk that made it clear, for example?

Reply:
cross linking to the exact sample conversation that lead to this: https://docs.google.com/document/d/1IWcY2hEyW5SWPJtp7lrWFDVmeG4HZkPidDlu_i1dR_8/edit?disco=AAABMmZLY-w

anyway sounds like we need to fix this particular case by fixing rag behavior, not by giving the llm more background in the system instructions (because again, that won't scale if we add another line to the system instructions for every thing the llm doesn't seem to know about openshift)

Reply:
+1, this one is odd, this is an experiment to see what knobs can be turned to get the desired effect.